core/evm.go

```diff
@@ -63,14 +63,17 @@ func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common
 	}
 
 	// Set goldTokenAddress
-	caller := &CeloBackend{config, statedb.Copy()}
-	registry, err := abigen.NewRegistryCaller(contracts_config.RegistrySmartContractAddress, caller)
-	if err != nil {
-		log.Error("Failed to access registry!", "err", err)
-	}
-	goldTokenAddress, err := registry.GetAddressForOrDie(&bind.CallOpts{}, contracts_config.GoldTokenRegistryId)
-	if err != nil {
-		log.Error("Failed to get address for GoldToken!", "err", err)
+	var goldTokenAddress common.Address
+	if config.IsCel2(header.Time) {
+		caller := &CeloBackend{config, statedb.Copy()}
```

Comment:
this seems like an expensive call... wonder if we should pay attention to this, and when possible do it only once and reuse for all our smart contract calls... (as we do static calls that don't modify state)...

Also, i wonder where the hot cache lives, if it's the state... then more arguments to keep it

Reply:
It should be possible to reuse the state copy for reading the whitelist and the exchange rates for the gas currencies. I'll try that as part of https://github.com/celo-org/optimism/issues/58.